mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1047,6 +1076,12 @@ private[ml] object RandomForest extends Logging {
       numNodesInGroup += 1
       memUsage += nodeMemUsage
     }
+    if (memUsage > maxMemoryUsage && maxMemoryUsage != 0) {
```

Comment:
I saw that you had simplified it using that trick, very clever! I don't think it's a big issue either way.